core/types/rollup_cost.go

```diff
@@ -124,6 +124,9 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 	forBlock := ^uint64(0)
 	var cachedFunc l1CostFunc
 	selectFunc := func(blockTime uint64) l1CostFunc {
+		if config.IsCel2(blockTime) {
```

Comment:
If we only enable this on Alfajores, we can avoid a hardfork and still have passing tests in the optimism repo.
```suggestion
		if config.IsCel2(blockTime) && config.ChainID.Cmp(44787) == 0 {
```

Reply:
I don't totally like the idea of tying the change for only a network, but I completely agree that this will make our life easier.
If we choose this, I would recommend to have this check as a specific function and that function have a big comment explaining that 44787 is alfajores, and why we did this.

```suggestion
		if isL1CeloCostSpecRequired(config, blockTime) {
```

and then create that function
```
// This function was created to avoid bla bla bla in Alfajores bla bla bla
func isL1CeloCostSpecRequired(config *params.ChainConfig, blockTime uint64) bool {
   return config.IsCel2(blockTime) && config.ChainID.Cmp(44787) == 0
}
```

Reply:
I totally agree about the comment. I don't mind having the check inline or pulled into a separate func.

I also agree that tying changes to networks is ugly, but it seems the lesser evil compared to skipping tests or having constantly failing CI. And I didn't think of a better solution so far.